keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -402,182 +393,262 @@ The following new `AdminNetworkPolicy` API will be added to the `policy.networki
 type AdminNetworkPolicyPeer struct {
-  NamespaceSetSelector         *NamespaceSet
-  NamespaceSetAndPodSelector   *NamespaceSetAndPod
+  Namespaces       *NamespaceSet
+  NamespacedPods   *NamespaceAndPodSet
 }
 
 // NamespaceSet defines a flexible way to select Namespaces in a cluster.
+// Exactly one of the selectors should be set.  If a consumer observes none of 
+// its fields are set, they should assume an option they are not aware of has 
+// been specified and fail closed.
 type NamespaceSet struct {
-  Scope               NamespaceMatchType
-  // Labels should be set only when scope is "SameLabels" or "NotSameLabels.
-  // +optional
-  Labels              []string
-  // NamespaceSelector should be set only when scope is "Selector". 
+  // Self cannot be "false" when it is set.  
+  // If Self is "true" then all pods in the subject's namespace are selected.
+  Self                *bool
+  // NotSelf cannot be "false" when it is set.  
+  // if NotSelf is "true" then all pods not in the subject's Namespace are selected.
+  NotSelf             *bool
+  // NamespaceSelector is a labelSelector used to select Namespaces, This field 
+  // follows standard label selector semantics; if present but empty, it selects 
+  // all Namespaces. 
   NamespaceSelector   *metav1.LabelSelector
+  // SameLabels is used to select a set of Namespaces that share the same label(s).  
+  // To be selected a Namespace must have all of the labels defined in SameLabels, 
+  // If Samelabels is Empty then nothing is selected
+  SameLabels          []string
+  // NotSameLabels is used to select a set of Namespaces that do not have a set 
+  // of label(s). To be selected a Namespace must have none of the labels defined 
+  // in NotSameLabels. If NotSameLabels is empty then nothing is selected
+  NotSameLabels       []string
 }
 
-// NamespaceSetAndPod defines a way to select Namespaces and Pods in a cluster 
-type NamespaceSetAndPod struct {
-  Scope               NamespaceMatchType
-  // Labels should be set only when scope is "SameLabels" or "NotSameLabels.
-  // +optional
-  Labels              []string
-  // NamespaceSelector should be set only when scope is "Selector". 
-  NamespaceSelector   *metav1.LabelSelector
-  // PodSelector should be set only when scope is "Selector", "Self", or "NotSelf".
-  // This allows the user to select matching pods in the same namespace
-  PodSelector         *metav1.LabelSelector 
+// PodSet defines a flexible way to select pods in a cluster. Exactly one of the 
+// selectors should be set.  If a consumer observes none of its fields are set, 
+// they should assume an option they are not aware of has been specified and fail closed.
+type PodSet struct { 
+  // PodSelector is a labelSelector used to select Pods, This field 
+  // follows standard label selector semantics; if present but empty, it selects 
+  // all Pods.  
+  PodSelector     *metav1.LabelSelector   
```

Comment:
Same as above - just `Selector` ?  But then, if we ever add a ServiceAccount, it would be clunky.  OK.  If we leave this `PodSelector`, we probably should leave `NamespaceSelector` .  More comment up there

Reply:
Well to add a new selector you would need to add a new struct like `ServiceAccountSet` so it actually wouldn't interfere here.

Reply:
done